core/dbt/exceptions.py

```diff
@@ -1019,6 +1019,17 @@ def get_message(self) -> str:
         return msg
 
 
+class MacroResultAlreadyLoadedError(CompilationError):
+    def __init__(self, result_name):
+        self.result_name = result_name
+        super().__init__(msg=self.get_message())
+
+    def get_message(self) -> str:
+        msg = f'The result "{self.result_name}" has already been loaded into a variable'
```

Comment:
Let's mention the word `statement` explicitly, since that should give users an extra hint. We're pretty inconsistent about how we quote variables in our exception/logging messages today. I tend to opt for single quotes or backticks.
```suggestion
        msg = f"The 'statement' result named '{self.result_name}' has already been loaded into a variable"
```